ooasp/encodings/ooasp_smart_generation.lp

```diff
@@ -0,0 +1,110 @@
+% =====================
+% Base program
+% =====================
+
+
+% Target associations are those with cardinality 1..1-MIN..MAX
+% We only consider this type of associations for the bound calculations
+is_target_assoc(ASSOC,SIDE):-
+    ooasp_assoc_limit(ASSOC,max,SIDE,C1,1,C2),
+    ooasp_assoc_limit(ASSOC,min,SIDE,C1,1,C2).
+
+% =====================
+% Incremental generation
+% =====================
+#program domain(new_object, cls).
+
+% #heuristic object_needed(ID1, ASSOC, A, C2, SIDE, new_object):object_needed(ID1, ASSOC, A, C2, SIDE, new_object),ooasp_assoc_specialization(ASSOC,_). [1,sign]
```

Comment:
Please don't include commented-out code; or at least add an explanation why it is commented-out.

Reply:
Just removed the line.